packages/mobile/src/transactions/TransferFeedIcon.tsx

```diff
@@ -0,0 +1,58 @@
+import ContactCircle from '@celo/react-components/components/ContactCircle'
+import RewardIcon from '@celo/react-components/icons/RewardIcon'
+import * as React from 'react'
+import { Image, StyleSheet, View } from 'react-native'
+import { coinsIcon, unknownUserIcon } from 'src/images/Images'
+import { getRecipientThumbnail, Recipient } from 'src/recipients/recipient'
+import { TransactionTypes } from 'src/transactions/reducer'
+
+const avatarSize = 40
+
+interface Props {
+  type: TransactionTypes
+  recipient?: Recipient
+  address?: string
+}
+
+export default function TransferFeedIcon(props: Props) {
```

Comment:
This approach is good but you may be able to further DRY up the type checking by having your new `getTransferFeedParams` helper return the icon based on the type instead of making a separate component here

Reply:
I looked into that approach but found that this was cleaner because so many TransactionTypes share the `coinsIcon` but require different cases in `getTransferFeedParams`

Reply:
👌